gateway/sds_gateway/api_methods/management/commands/reset_index.py

```diff
@@ -0,0 +1,494 @@
+                )
+            )
+
+            # Check for new fields and apply transforms
+            new_fields = self.get_new_fields(client, source_index, dest_index)
+            if new_fields:
+                self.stdout.write(f"Found new fields: {new_fields}")
+                self.apply_field_transforms(client, dest_index, new_fields)
+
+        except (RequestError, OpensearchConnectionError, NotFoundError) as e:
+            self.stdout.write(self.style.ERROR(f"Error during reindex: {e!s}"))
+            if isinstance(e, RequestError) and e.info.get("failures"):
+                self.stdout.write(
+                    self.style.ERROR(
+                        f"Some documents failed to reindex: {e.info['failures']}"
+                    )
+                )
+
+                # prompt input to reindex the failed documents
+                manual_reindex = (
+                    input("Reindex failed documents manually? (y/N): ").lower() == "y"
+                )
+                self.stdout.write(
+                    self.style.WARNING(f"Manual reindex: {manual_reindex}")
+                )
+                if manual_reindex:
+                    self.stdout.write(
+                        self.style.WARNING("Reindexing failed documents manually...")
+                    )
+
+                    # manually index the failed documents, skipping deleted captures
+                    for failure in e.info["failures"]:
+                        capture = Capture.objects.get(uuid=failure["id"])
+                        if not capture.is_deleted:
+                            self.reindex_single_capture(capture)
+
+    def get_doc_count(self, client: OpenSearch, index_name: str) -> int:
+        """Get the number of documents in an index."""
+        try:
+            return client.count(index=index_name)["count"]
+        except NotFoundError:
+            self.stdout.write(self.style.ERROR(f"Index '{index_name}' not found"))
+            return -1
+        except OpensearchConnectionError as e:
+            self.stdout.write(
+                self.style.ERROR(
+                    f"Connection error while accessing '{index_name}': {e!s}"
+                )
+            )
+            return -1
+        except AuthenticationException as e:
+            self.stdout.write(
+                self.style.ERROR(f"Authentication failed for '{index_name}': {e!s}")
+            )
+            return -1
+        except (RequestError, ValueError) as e:
+            self.stdout.write(
+                self.style.ERROR(
+                    f"Unexpected error getting count for '{index_name}': {e!s}"
+                )
+            )
+            return -1
+
+    def handle(self, *args, **options):
+        """Execute the command."""
+        client: OpenSearch = get_opensearch_client()
+        capture_type = options["capture_type"]
+        index_name = options["index_name"]
+
```

Comment:
let's add a manual confirmation in red / error-style to warn this is a potentially destructive operation on existing Captures and OpenSearch indices. Bail out by default, or when the user doesn't reply with `y`.